sdk/src/cameras/itof-frame/frame_impl.cpp

```diff
@@ -255,3 +255,26 @@ aditof::Status FrameImpl::getAttribute(const std::string &attribute,
     value = m_attributes[attribute];
     return aditof::Status::OK;
 }
+
+aditof::Status FrameImpl::getTemperature(uint32_t &sensorTemp,
+                                         uint32_t &laserTemp) const {
+
+    aditof::Status status = aditof::Status::OK;
+    uint8_t *header;
+
+    if (m_implData->m_dataLocations.count("embedded_header") > 0) {
+        header = reinterpret_cast<uint8_t *>(
+            m_implData->m_dataLocations["embedded_header"]);
+    } else {
+        LOG(ERROR) << "Metadata is not enabled in this frame!";
```

Comment:
Not really an error. Users can configure adsd3500 to not generate metadata.
I would log anything here.

Reply:
Fixed.